src/ir/module.h

```diff
@@ -34,11 +34,13 @@ class Block;
 class Operation {
  public:
   Operation(const Block* parent, const Operator& op,
-            NamedAttributeMap attributes, NamedValueMap inputs)
+            NamedAttributeMap attributes, NamedValueMap inputs,
+            std::unique_ptr<Module> module = nullptr)
```

Comment:
My first thought was that, aesthetically, it's a little odd to have the `parent` `Block` be the first parameter and the parent `Module` be the last parameter. Then I realized that this `Module` is not the parent, but the implementation. To reduce confusion, maybe name the module something like `impl_module`?

Reply:
Good idea. Will fix it.  

I think there is a related problem with the use of `module` in `Block`. It should really be called `parent` there. I will fix that issue in a separate PR.

Reply:
Done.